acl.go

```diff
@@ -220,13 +233,6 @@ func (odbi *ovndb) aclDelImp(lsw, direct, match string, priority int, external_i
 		row["priority"] = priority
 	}
 
-	if external_ids != nil {
```

Comment:
Sorry that I have a second thought on the ACLDel interface & implementation while reviewing this code. It seems the arg external_ids can be removed. However, I want to avoid interface change if possible. On the other hand, it seems confusing for users if the arg is provided by not used. I think it should be honored if not nil - the API user should have a natural expectation that the external_ids is used for looking up the candidate for deletion when they pass in a non-nil value. And it is not harmful to support it, because users can always put a nil value if they don't want it to be part of the comparison. What do you think?

Reply:
I agree that we can't remove external_ids from ACLDel() because we need to preserve backward compatibility, and I also agree that this could be confusing to the user.  However, I think it could also be confusing to support external_ids as an ACLDel() match criterion.  The way ACLAdd works now, there can only be one ACL for a given entity (logical switch or port group), direction, match, and priority.  I don’t see any added value in optionally supporting external_ids as a match criterion, and if we do support it, and they provide the wrong set of external_ids, the ACL won’t get deleted and we’ll return an error.

Vasiliy mentioned on Slack the possibility of a new API that “search only via uniq field and another via matched fields. So technically you can delete for example all ACLS with specific fields in external ids.“ 

If I understand correctly, I think this could be implemented via a new API called, for example, “ACLDelBulk()”.  ACLDelBulk() would delete all ACLs that matched the provided criteria.  I can imagine a use case where we have a set of ACLs that implement a policy.  We could tag them all via external_ids with the policy name, and then delete them all with one call if the policy is changed.  I don’t think we have a use case for that right now, but I’m open to implementing that functionality separately if needed.

My proposal is the following:
1. Add a comment on the API in my next PR (ACL for port groups) saying that external_ids are no longer used, but are being retained for backward compatibility. The user can safely use nil for the field. Note, I already have a comment in my next PR saying that the ACLDel() API is being deprecated in favor of ACLDelEntity().
2. Remove external_ids my new ACLDelEntity() API.
3. Remove external_ids from aclDelImp() so if the user looks into it, they will see pretty quickly and clearly that it’s not used.

What do you think?

Reply:
> and if we do support it, and they provide the wrong set of external_ids, the ACL won’t get deleted and we’ll return an error.

I think in this case if they provide a wrong set of external_ids, then the expectation to the API should be fail instead of deleting by mistake.

For the idea of ACLDelBulk(), I agree, but can be discussed separately.